### ISSUE TITLE
query with join and vector search

### ISSUE THREAD

**Issue by ruslandoga** [typesense/typesense#1164]
## Description

I have two schemas

```json
{
  "default_sorting_field": "recent_downloads",
  "fields": [
    {
      "facet": true,
      "name": "name",
      "type": "string"
    },
    {
      "name": "recent_downloads",
      "type": "int32"
    },
    {
      "name": "vec",
      "num_dim": 64,
      "optional": true,
      "type": "float[]"
    }
  ],
  "name": "packages",
  "symbols_to_index": [
    "_"
  ]
}
```
```json
{
  "fields": [
    {
      "facet": true,
      "name": "package",
      "reference": "packages.name",
      "type": "string"
    },
    {
      "index": false,
      "name": "ref",
      "optional": true,
      "type": "string"
    },
    {
      "facet": true,
      "name": "type",
      "type": "string"
    },
    {
      "infix": true,
      "name": "title",
      "type": "string"
    }
  ],
  "name": "docs",
  "token_separators": [
    "."
  ]
}
```

and I'd like to join `docs` with `packages` in a `multi_search` query like

```json
{
  "searches": [
    {
      "collection": "docs",
      "filter_by": "$packages(name:=package)",
      "infix": "always",
      "q": "all",
      "query_by": "title",
      "sort_by": "_text_match(buckets: 3):desc,_vector_distance:asc,$packages(recent_downloads:desc)",
      "vector_query": "$packages(vec:([0.2135181576013565,-0.5124136805534363,-0.28749880194664,0.12114500999450684,-0.09998839348554611,-0.10657893866300583,0.32304078340530396,0.06316358596086502,-0.2526101768016815,0.1981559544801712,0.3690982162952423,-0.07460297644138336,-0.2819271385669708,-0.15467767417430878,-0.16187314689159393,0.33524030447006226,-0.1485835313796997,0.026179572567343712,-0.01642446219921112,0.30825477838516235,0.4877499043941498,-0.15968985855579376,0.5061681270599365,-0.4759694039821625,0.14635320007801056,-0.21382741630077362,-0.3524484634399414,-0.10986503958702087,0.3730050027370453,-0.22404232621192932,0.2498970329761505,-0.29987749457359314,0.1399274468421936,0.11405358463525772,-0.10211364924907684,-0.06355232000350952,0.6305976510047913,0.1747753918170929,0.13079020380973816,0.22149261832237244,0.2953811287879944,0.28434130549430847,0.20399589836597443,-0.24646367132663727,0.07994255423545837,0.2164379507303238,0.1539556384086609,-0.32746872305870056,-0.037217117846012115,-0.5059004426002502,0.21480853855609894,0.005424731411039829,0.25962573289871216,-0.0911545604467392,0.5147342085838318,0.39061230421066284,-0.09368181228637695,-0.09403053671121597,0.42599377036094666,0.45821478962898254,0.1998760849237442,0.28837576508522034,-0.09650873392820358,0.30293533205986023], k:64))"
    }
  ]
}
``` 

## Expected Behavior

That query joins `docs` on `packages` and performs vector query on `packages.vec` and sorts by `packages.recent_downloads`

## Actual Behavior

I'm getting various errors like

> Failed to apply reference filter on `packages` collection: Could not find any field in `packages` referencing the collection `docs`.

> Could not find a field named `$packages(recent_downloads` in the schema for sorting.

etc.

## Metadata

**Typesense Version**: `typesense/typesense:0.25.0`

**OS**: Docker on Mac

**Comment by kishorenc** [comment]
@ruslandoga 

The join feature is being actively developed on the 0.26 branch. It's not available as part of 0.25.0. The latest 0.26 RC build is `0.26.0.rc14` -- please try with that and let me know if it works.

**Comment by jasonbosco** [comment]
The JOIN feature is now available in v26.
